EngineMath/Vector.cpp

```diff
@@ -8,4 +8,76 @@
 
 namespace EngineMath {
 
+	//Public CTORS
+	Vector::Vector() {
+		data = new T[size];
+		for (int i = 0; i < size; i++)
+			data[i] = 0;
+	}
+
+	Vector::Vector(Ts... args) {
+		data = new T[size];
+		data = {args...};
+	}
+
+	//Public Methods
+
+	Vector::Dot(const Vector& other) {
```

Comment:
1. This function has no return type. It will not compile.
2. There's no real way to error out of this Dot product without throwing an exception.
    - Exceptions are a no-go for this project, they unwind the call stack which make debugging difficult in event of an error, and have some pretty major performance hits if used incorrectly
    -  There's no "error" value you could reasonably return that wouldn't be a value of some arbitrary dot product


The only real answers to 2 are:
- Use exceptions
- Redesign so that this error is caught at compile time

I strongly prefer the latter.